src/OpenApi/Factory/OpenApiFactory.php

```diff
@@ -158,7 +158,11 @@ private function collectPaths(ApiResource $resource, ResourceMetadataCollection
 
             [$requestMimeTypes, $responseMimeTypes] = $this->getMimeTypes($operation);
 
-            $operationId = $operation->getOpenapiContext()['operationId'] ?? $this->normalizeOperationName($operationName);
+            $operationId = $operation->getOpenapi()?->getOperationId() ?: $this->normalizeOperationName($operationName);
```

Comment:
we need to handle when `getOpenapi` is `true`, I guess `false` is handled above. Maybe that we could just propagate the `openApiOperation` if it exists to underlying functions?